webrender/src/prim_store.rs

```diff
@@ -580,6 +566,74 @@ impl PrimitiveStore {
         PrimitiveIndex(prim_index)
     }
 
+    pub fn resolve_primitives(&mut self, resource_cache: &ResourceCache) {
+        for prim_index in self.prims_to_resolve.drain(..) {
+            let metadata = &mut self.cpu_metadata[prim_index.0];
+
+            if let Some(mask_key) = metadata.mask_image {
+                let tex_cache = resource_cache.get_image(mask_key, ImageRendering::Auto);
+                metadata.mask_texture_id = tex_cache.texture_id;
+                if let Some(address) = metadata.clip_index {
+                    let clip = self.gpu_data32.get_slice_mut(address, 6);
+                    let old = clip[5].data; //TODO: avoid retaining the screen rectangle
+                    clip[5] = GpuBlock32::from(ImageMaskInfo {
```

Comment:
At some point I'd like to factor all these indices out into constants. (You don't have to do this now)

Maybe we could even make macros to make this stuff easier.